### PR TITLE
Fix more BBEs

### PR DESCRIPTION
**Pull request by irshadnilam**
## Purpose
$title

### CODE REVIEW

File: examples/checkpanic/checkpanic.out [hunk 1]
```diff
@@ -2,7 +2,4 @@
 # `.bal` file, and execute the `ballerina run` command.
 $ ballerina run checkpanic.bal
 120
-error: {ballerina}ConversionError {"message":"incompatible convert operation: 'string' value 'Invalid' cannot be converted as 'int'"}
-        at ballerina/utils:simpleValueConvert(<native>)
-           parse(checkpanic.bal:5)
-           main(checkpanic.bal:19)
+error: invoking main method failed due to error {ballerina}NumberParsingError message='string' value 'Invalid' cannot be converted to 'int'
```
Comment (MaryamZi) [inline examples/checkpanic/checkpanic.out]: This again seems to be a bug (https://github.com/ballerina-platform/ballerina-lang/issues/17072), let's not change this.

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 2]
```diff
@@ -1,37 +1,33 @@
 import ballerina/io;
 
-type SampleError error<string, map<anydata|error>>;
+type SampleErrorData record {
+    string detail;
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]: Let's change this field to something else. A `detail` field in a detail mapping could get confusing.

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 3]
```diff
@@ -1,37 +1,33 @@
...
 public function main() {
     // This error-type binding pattern will destructure an `error` of the type `SampleError`, and create two variables as follows:
     // The value of the reason string in the`SampleError` will be set to a new `string` variable `reason`.
-    // The value of the detail mapping will be set to a new `map<anydata|error>` variable `detail`.
-    SampleError error(reason, detail) = getSampleError();
+    // The value of the detail mapping will be set to a specified variables.
+    var error(reason, detail = detail, fatal = fatal) = getSampleError();
     io:println("Reason String: " + reason);
-    io:println(io:sprintf("Detail Mapping: %s", detail));
+    io:println(io:sprintf("Detail string: %s", detail));
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]: Not sure why we've used `sprintf` here. We should be able to just do 
```suggestion
    io:println("Detail string: ", detail);
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 4]
```diff
@@ -1,37 +1,33 @@
...
-
-    // Error-type binding patterns can be used with `var` to infer the type from the right hand side.
-    // Since the types of the new variables are based on the type of the type-binding pattern, using `var` will
-    // infer the types from the right hand side.
-    var error(vReason, vDetail) = getSampleError();
-    // Type of `vReason` is inferred as `string`.
-    io:println("Reason String: " + vReason);
-    // Type of `vDetail` is inferred as `map<anydata|error>`.
-    io:println(io:sprintf("Detail Mapping: %s", vDetail));
+    io:println(io:sprintf("Detail map: %s", params));
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]: Let's add a comment about what goes into params.

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 5]
```diff
@@ -1,37 +1,33 @@
...
     // The underscore '_' sign can be used to ignore either the reason string or the detail mapping.
-    error<string, Foo> error(_, fooRec) = getRecordConstrainedError();
-    io:println(io:sprintf("Detail Mapping: %s", <Foo> fooRec));
+    var error(_, detailMsg = detailMsg, isFatal = isFatal) = getRecordConstrainedError();
+    io:println(io:sprintf("Detail message: %s", detailMsg));
 }
 
 function getSampleError() returns SampleError {
-    SampleError e = error("Sample Error", { detail: "Detail Msg", fatal: true });
+    SampleError e = error("Sample Error", detail = "Detail Msg", fatal = true );
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    SampleError e = error("Sample Error", detail = "Detail Msg", fatal = true);
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 6]
```diff
@@ -41,6 +37,6 @@ type Foo record {|
 |};
 
 function getRecordConstrainedError() returns error<string, Foo> {
-    error<string, Foo> e = error("Some Error", { detailMsg: "Failed Message", isFatal: true });
+    error<string, Foo> e = error("Some Error", detailMsg = "Failed Message", isFatal = true );
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    error<string, Foo> e = error("Some Error", detailMsg = "Failed Message", isFatal = true);
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 7]
```diff
@@ -1,37 +1,37 @@
 import ballerina/io;
 
-type SampleError error<string, map<anydata|error>>;
+type SampleErrorData record {
+    string message?;
+    error cause?;
+    string info;
+    boolean fatal;
+};
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
};

```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 8]
```diff
@@ -1,37 +1,37 @@
...
 public function main() {
     // This error-type binding pattern will destructure an `error` of the type `SampleError`, and create two variables as follows:
     // The value of the reason string in the`SampleError` will be set to a new `string` variable `reason`.
-    // The value of the detail mapping will be set to a new `map<anydata|error>` variable `detail`.
-    SampleError error(reason, detail) = getSampleError();
+    // The value of the detail mapping will be set to a specified variables.
+    var error(reason, info = info, fatal = fatal) = getSampleError();
     io:println("Reason String: " + reason);
-    io:println(io:sprintf("Detail Mapping: %s", detail));
+    io:println("Info string: ", info);
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    io:println("Info: ", info);
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 9]
```diff
@@ -1,37 +1,37 @@
...
+    io:println("Is fatal: ", fatal);
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    io:println("Fatal: ", fatal);
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 10]
```diff
@@ -1,37 +1,37 @@
...
 
-    // The detail-mapping value can further be destructured using a record-binding pattern.
-    SampleError error(reasonTwo, { detail: detailTwo, fatal }) = getSampleError();
+    // The detail-mapping value can also be destructured using a rest parameter.
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    // The detail mapping can also be destructured using a rest parameter.
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 11]
```diff
@@ -1,37 +1,37 @@
...
+    // `params` will be of type `map<anydata|error>`, and will have the values of
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    // `params` will be of type `map<anydata|error>`, and will have the 
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 12]
```diff
@@ -1,37 +1,37 @@
...
+    // `info` and `fatal` fields.
+    var error(reasonTwo, ...params) = getSampleError();
     io:println("Reason String: " + reasonTwo);
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    io:println("Reason String: ", reasonTwo);
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 13]
```diff
@@ -1,37 +1,38 @@
...
+};
+
+type SampleError error<string, SampleErrorData>;
 
 public function main() {
     // This error-type binding pattern will destructure an `error` of the type `SampleError`, and create two variables as follows:
     // The value of the reason string in the`SampleError` will be set to a new `string` variable `reason`.
-    // The value of the detail mapping will be set to a new `map<anydata|error>` variable `detail`.
-    SampleError error(reason, detail) = getSampleError();
+   // The values in the detail mapping will be set to the specified variables.
```
Comment (MaryamZi) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]: These are new variables right?

```suggestion
   // The values in the detail mapping will be set to new variables `info` and `fatal`.
```

Comment (irshadnilam) [reply examples/error-type-binding-pattern/error_type_binding_pattern.bal]: yes, type_binding

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 14]
```diff
@@ -1,37 +1,38 @@
...
+};
+
+type SampleError error<string, SampleErrorData>;
 
 public function main() {
     // This error-type binding pattern will destructure an `error` of the type `SampleError`, and create two variables as follows:
     // The value of the reason string in the`SampleError` will be set to a new `string` variable `reason`.
-    // The value of the detail mapping will be set to a new `map<anydata|error>` variable `detail`.
-    SampleError error(reason, detail) = getSampleError();
+    // The values in the detail mapping will be set to new variables `info` and `fatal`.
```
Comment (praneesha) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    // The values in the detailed mapping will be set to the new variables `info` and `fatal`.
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 15]
```diff
@@ -1,37 +1,38 @@
...
-    // infer the types from the right hand side.
-    var error(vReason, vDetail) = getSampleError();
-    // Type of `vReason` is inferred as `string`.
-    io:println("Reason String: " + vReason);
-    // Type of `vDetail` is inferred as `map<anydata|error>`.
-    io:println(io:sprintf("Detail Mapping: %s", vDetail));
+    io:println("Info: ", info);
+    io:println("Fatal: ", fatal);
+
+    // The detail mapping can also be destructured using a rest parameter.
```
Comment (praneesha) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    // The detailed mapping can also be destructured using a rest parameter.
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal [hunk 16]
```diff
@@ -1,37 +1,38 @@
...
+    // `params` will be of type `map<anydata|error>`, and will have the
```
Comment (praneesha) [inline examples/error-type-binding-pattern/error_type_binding_pattern.bal]:
```suggestion
    // `params` will be of the type `map<anydata|error>` and will have the 
```

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal (hunk 14 above)
Comment (irshadnilam) [reply examples/error-type-binding-pattern/error_type_binding_pattern.bal]: Error object has two fields called "reason" and "detail". The "detail" result (mapping) is stated here.

File: examples/error-type-binding-pattern/error_type_binding_pattern.bal (hunk 16 above)
Comment (irshadnilam) [reply examples/error-type-binding-pattern/error_type_binding_pattern.bal]: Error object has two fields called "reason" and "detail". The "detail" result (mapping) is stated here.

**Pull request merged**